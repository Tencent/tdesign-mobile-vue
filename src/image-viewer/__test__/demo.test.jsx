/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/image-viewer/demos/base.vue';
import mobileVue from '@/image-viewer/demos/mobile.vue';
import operationVue from '@/image-viewer/demos/operation.vue';

const mapper = {
  baseVue,
  mobileVue,
  operationVue,
};

describe('ImageViewer', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`ImageViewer ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
