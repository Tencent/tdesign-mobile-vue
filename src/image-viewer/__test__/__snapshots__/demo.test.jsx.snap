// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ImageViewer > ImageViewer baseVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      基础图片预览
      
    </span>
    <!---->
  </button>
  <!---->
  
</div>
`;

exports[`ImageViewer > ImageViewer mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
  data-v-ef8302fa=""
>
  <h1
    class="title"
    data-v-ef8302fa=""
  >
    ImageViewer 图片预览
  </h1>
  <p
    class="summary"
    data-v-ef8302fa=""
  >
    图片全屏放大预览效果，包含全屏背景色、页码位置样式、增加操作等规范
  </p>
  <div
    class="tdesign-mobile-demo-block"
    data-v-ef8302fa=""
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        01 组件类型
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        图片预览类型
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot with-padding"
    >
      
      
      <button
        aria-disabled="false"
        class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
        role="button"
        type="button"
      >
        <!---->
        <span
          class="t-button__content"
        >
          
          基础图片预览
          
        </span>
        <!---->
      </button>
      <!---->
      
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block"
    data-v-ef8302fa=""
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        02 组件类型
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        图片预览类型，可设置垂直对齐方式
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot with-padding"
    >
      
      
      <button
        aria-disabled="false"
        class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
        role="button"
        type="button"
      >
        <!---->
        <span
          class="t-button__content"
        >
          
           基础图片预览 + 对齐方式 
          
        </span>
        <!---->
      </button>
      <!---->
      
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block"
    data-v-ef8302fa=""
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        03 组件类型
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        带操作图片预览
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot with-padding"
    >
      
      
      <button
        aria-disabled="false"
        class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
        role="button"
        type="button"
      >
        <!---->
        <span
          class="t-button__content"
        >
          
          带操作图片预览
          
        </span>
        <!---->
      </button>
      <!---->
      
      
    </div>
  </div>
</div>
`;

exports[`ImageViewer > ImageViewer operationVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <button
    aria-disabled="false"
    class="t-button t-button--size-large t-button--outline t-button--primary t-button--rectangle t-button--block"
    role="button"
    type="button"
  >
    <!---->
    <span
      class="t-button__content"
    >
      
      带操作图片预览
      
    </span>
    <!---->
  </button>
  <!---->
  
</div>
`;
